app/assets/javascripts/findingaids.js.coffee

```diff
@@ -1,4 +1,5 @@
 $ ->
   # Hide search fields if url specifies repository
   if $("#repository_header").html() != ""
-    $("#search_field").hide()
+    $("#search_field option[value='" + $('#repository_header').text().replace(': ', '') + "']").attr("selected","selected");
```

Comment:
@chrpr You actually do not need this, setting `:search_field` in the route will preselect this. You'll see that it works for all the libraries except the ones with ampersands because there is a URL encoding clash. So there are a few options here: to keep this functionality you could add something like the following the repositories YAML:

```
url_safe_display: 'The Fales Library %26 Special Collections'
```

And then make that the value of `:search_field` both in the route and in the catalog_controller where you're setting `add_search_field`.

But looking through this functionality there is an inherent problem that I believe has been causing what seems like a bug (but it's actually a feature :smile: ): When you are at any stable repos url, e.g. `/rism`, the `:search_field` will always be set to 'Research Institute for the Study of Man'. This means that if you select Fales from the dropdown then it will still search RISM.

To fix this we will need to use the coffee but we need to allow it to be changed based on selection. The way you have it now forces it to always be selected based on the header text. So something like:

```
if $("#repository_header").html() != "" && $( "#search_field option:selected" ).text() == "All Libraries"
    $("#search_field option[value='" + $('#repository_header').text().replace(': ', '') + "']").attr("selected","selected");
```